web3.js/src/util/send-and-confirm-transaction.ts

```diff
@@ -33,18 +34,36 @@ export async function sendAndConfirmTransaction(
     sendOptions,
   );
 
-  const status = (
-    await connection.confirmTransaction(
-      signature,
-      options && options.commitment,
-    )
-  ).value;
+  const status = await connection.getSignatureStatus(signature);
 
-  if (status.err) {
-    throw new Error(
-      `Transaction ${signature} failed (${JSON.stringify(status)})`,
+  const checkBlockHeight = async () => {
+    const blockHeight = await connection.getBlockHeight(
+      options && options.commitment,
     );
```

Comment:
Whatever we do, we need a way to be able to ignore the result of this function in case we learn, in the meantime, that the signature has confirmed. You never want to have learned that the signature confirmed, but _also_ be waiting on this RPC call to complete before notifying your application.

Maybe you could accomplish this with _two_ long-running promises, each with their own `AbortController`, and `Promise.race`.

Reply:
We talked about this for a bit last night, and I think we understand what you're asking for but we haven't gotten anything concrete complete.

Reply:
I spent some time working on this tonight, and I think I have something closer to what you had in mind. Wouldn't mind another quick look to see what you think of the updates.